baseplate/url.py

```diff
@@ -0,0 +1,32 @@
+"""URL domain parsing."""
+
+
+import re
+
+DOMAIN_REGEX = re.compile("(?i)(?:.+?://)?([^/:#?]*)")
+DOMAIN_PREFIX_REGEX = re.compile(r"^www\d*\.")
+
+
+def is_subdomain(subdomain: str, base: str) -> bool:
```

Comment:
do we maybe want to define a Domain [newtype](https://docs.python.org/3/library/typing.html#newtype) to tie these functions together?

```
Domain = NewType("Domain", str)
get_domain(url: str) -> Domain
is_subdomain(subdomain: Domain, base: Domain) -> bool
```

or maybe that makes it too annoying to use these with regular strings?

Reply:
That seems like overkill to me?

Reply:
agreed, i've got types on the brain :)